client/test/commands/userInputUtil.test.ts

```diff
@@ -1257,4 +1257,62 @@ describe('userInputUtil', () => {
         });
     });
 
+    describe('saveDirectory', () => {
+        it('should finish if user doesnt select browse', async () => {
+            const placeHolder: string = 'Choose the location to save the smart contract';
+            const result: vscode.Uri = await UserInputUtil.saveDirectory(placeHolder);
+
+            quickPickStub.should.have.been.calledWith([UserInputUtil.BROWSE_LABEL], { placeHolder });
+
+            should.not.exist(result);
+        });
+
+        it('should allow folders to be chosen when selected, and not files', async () => {
+            quickPickStub.resolves(UserInputUtil.BROWSE_LABEL);
+            const showOpenDialogStub: sinon.SinonStub = mySandBox.stub(vscode.window, 'showOpenDialog').resolves([{ fsPath: '/some/path' }]);
+            const placeHolder: string = 'Choose the location to save the smart contract';
+            const result: vscode.Uri = await UserInputUtil.saveDirectory(placeHolder);
+
+            quickPickStub.should.have.been.calledWith([UserInputUtil.BROWSE_LABEL], { placeHolder });
+            showOpenDialogStub.should.have.been.calledWith({
+                canSelectFiles: false,
+                canSelectFolders: true,
+                canSelectMany: false,
+                openLabel: 'Save'
+            });
+
+            result.fsPath.should.equal('/some/path');
+        });
+
+        it('should cancel if no directory is selected', async () => {
+            quickPickStub.resolves(UserInputUtil.BROWSE_LABEL);
+            const showOpenDialogStub: sinon.SinonStub = mySandBox.stub(vscode.window, 'showOpenDialog').resolves(undefined);
+            const placeHolder: string = 'Choose the location to save the smart contract';
+            const result: vscode.Uri = await UserInputUtil.saveDirectory(placeHolder);
+
+            quickPickStub.should.have.been.calledWith([UserInputUtil.BROWSE_LABEL], { placeHolder });
+            showOpenDialogStub.should.have.been.calledWith({
+                canSelectFiles: false,
+                canSelectFolders: true,
+                canSelectMany: false,
+                openLabel: 'Save'
+            });
+
+            should.equal(result, undefined);
+        });
+
+        it('should handle any errors', async () => {
+            const logStub: sinon.SinonStub = mySandBox.stub(VSCodeBlockchainOutputAdapter.instance(), 'log');
```

Comment:
This should be a spy you don't need to stub it

Reply:
Done!